From 09a0f6e3b7f813bc96156dc4d0bcf9f4b17b7bad Mon Sep 17 00:00:00 2001
From: He Sheng <hesheng@wxiat.com>
Date: Fri, 16 Sep 2022 15:17:44 +0800
Subject: [PATCH] sw64: tools: add R_SW64_LITERAL_GOT support for relocs
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: 09a0f6e3b7f813bc96156dc4d0bcf9f4b17b7bad
Modified-by-SEL: No


Sunway inclusion
category: feature
bugzilla: https://gitee.com/openeuler/kernel/issues/I56OLG

--------------------------------

Signed-off-by: He Sheng <hesheng@wxiat.com>
Reviewed-by: Cui Wei <cuiwei@wxiat.com>
Signed-off-by: Gu Zitao <guzitao@wxiat.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 arch/sw_64/tools/relocs.c | 1 +
 arch/sw_64/tools/relocs.h | 1 +
 2 files changed, 2 insertions(+)

diff --git a/arch/sw_64/tools/relocs.c b/arch/sw_64/tools/relocs.c
index a8a9e08a0a65..06bd4625bc6e 100644
--- a/arch/sw_64/tools/relocs.c
+++ b/arch/sw_64/tools/relocs.c
@@ -487,6 +487,7 @@ static int do_reloc(struct section *sec, Elf_Rel *rel, Elf_Sym *sym,
 	case R_SW64_SREL32:
 	case R_SW64_GPRELHIGH:
 	case R_SW64_GPRELLOW:
+	case R_SW64_LITERAL_GOT:
 		/*
 		 * NONE can be ignored and PC relative relocations don't
 		 * need to be adjusted.
diff --git a/arch/sw_64/tools/relocs.h b/arch/sw_64/tools/relocs.h
index 37ac09ec2a77..17c7e31113a0 100644
--- a/arch/sw_64/tools/relocs.h
+++ b/arch/sw_64/tools/relocs.h
@@ -53,6 +53,7 @@
 #define R_SW64_TPRELHI		39
 #define R_SW64_TPRELLO		40
 #define R_SW64_TPREL16		41
+#define R_SW64_LITERAL_GOT	43	/* GP relative */
 
 void die(char *fmt, ...);
 
-- 
2.33.0

